src/components/pages/WikiNotePage.tsx

```diff
@@ -464,6 +469,13 @@ const WikiNotePage = ({ storage }: WikiNotePageProps) => {
     }
   }, [generalStatus.focusOnEditorCursor, hash, setGeneralStatus])
 
+  useEffect(() => {
+    if (noteId != null && note != null && noteId != previousNoteId) {
+      addVisitedToHistory(note)
+      setPreviousNoteId(noteId)
```

Comment:
Store previous note id state in ref hook rather than a state hook since we don't need to render it.
Also, it will prevent rendering the component twice.

Reply:
Done, will update PR later